pennylane/_qubit_device.py

```diff
@@ -543,13 +578,15 @@ def analytic_probability(self, wires=None):
         """
         raise NotImplementedError
 
-    def estimate_probability(self, wires=None):
+    def estimate_probability(self, wires=None, shot_range=None, bin_size=None):
         """Return the estimated probability of each computational basis state
         using the generated samples.
 
         Args:
             wires (Iterable[Number, str], Number, str, Wires): wires to calculate
                 marginal probabilities for. Wires not provided are traced out of the system.
+            shot_range (tuple[int]): 2-tuple of integers specifying the range of samples
+                to use. If not specified, all samples are used.
```

Comment:
Also add bin_size arg ?

Reply:
Oops, forgot a few docstring arguments I can see.

I'm still a bit wary of adding new kwargs, but it feels safer than attributes.